crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2031,6 +2038,33 @@ impl AuthorityPerEpochStore {
         Ok(())
     }
 
+    /// Notifies that a synced checkpoint of sequence number `checkpoint_seq` is available. The source of the notification
+    /// is the CheckpointExecutor.
+    pub fn notify_synced_checkpoint(&self, checkpoint_seq: CheckpointSequenceNumber) {
+        let mut highest_synced_checkpoint = self.highest_synced_checkpoint.write();
+        *highest_synced_checkpoint = checkpoint_seq;
```

Comment:
nit: can you leave a small comment that this is guaranteed to be notified in sequence order